src/main/kotlin/com/example/flashcardbackend/deck/DeckController.kt

```diff
@@ -11,10 +13,11 @@ class DeckController(val service: DeckService) {
 
     @GetMapping("/decks/{id}")
     fun getById(@PathVariable("id") id: Int): DeckDTO? {
-        return service.findDeckById(id)
+        return service.findDeckById(id) ?: throw(NotFoundException("Deck with id $id not found."))
```

Comment:
I think that in the last meeting I said that it doesn't matter which class throws the exception (service or controller), but I think that there are two cases when it should be thrown by the service:

1. You want to rollback the transaction. By default, Spring rollbacks the transaction if `@Transactional` method throws a `RuntimeException`.
2. The service has multiple "clients" and you want to ensure that every client gets the same behaviour when an error occurs (or finder method doesn't find anything).

Reply:
Thanks for the info!